awscrt/common.py

```diff
@@ -18,3 +18,23 @@ def get_cpu_count_for_group(group_idx: int) -> int:
     Returns number of processors in a given group.
     """
     return _awscrt.get_cpu_count_for_group(group_idx)
+
+class SystemEnvironment:
+
+    def __init__(self):
```

Comment:
Inherit from NativeResource. we just do this to help tests track leaks
```suggestion
class SystemEnvironment(NativeResource):

    def __init__(self):
        super().__init__()
```

Reply:
I'm just gonna pull out systemenvironment entirely and cache it internally as you suggested above.